www/src/components/MenuOverlayContainer/MenuOverlayContainer.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @emails react-core
+*/
+
+'use strict';
+
+import {Component} from 'react';
+import PropTypes from 'prop-types';
+import {media} from 'theme';
+
+class MenuOverlayContainer extends Component {
+  constructor(props) {
+    super(props);
+    this.toggleMenuOverlay = this.toggleMenuOverlay.bind(this);
+    this.handleWindowResize = this.handleWindowResize.bind(this);
+    this.state = {
+      open: false,
+    };
+  }
+  componentDidUpdate(prevProps, prevState) {
+    if (this.state.open) {
+      window.addEventListener('resize', this.handleWindowResize);
+    } else {
+      window.removeEventListener('resize', this.handleWindowResize);
+    }
+  }
+
+  componentWillUnmount() {
+    this.context.onMenuOverlayToggle(false);
+    window.removeEventListener('resize', this.handleWindowResize);
+  }
+
+  handleWindowResize() {
+    const mediaQuery = media.greaterThan('small').replace('@media ', '');
+    if (window.matchMedia(mediaQuery).matches) {
+      // If the screen has been resized >= 'small'
+      this.toggleMenuOverlay(false);
+    }
+  }
+
+  toggleMenuOverlay(willOpen) {
+    this.setState(
+      {
+        open: willOpen,
+      },
+      () => {
+        this.context.onMenuOverlayToggle(this.state.open);
+      },
+    );
+  }
+
+  handleTouchMove(evt) {
```

Comment:
This method needs to be bound to the instance. Else it errors during runtime:

1. Open the mobile nav
1. Click the "X" to close it
1. Touch and scroll the page to drag
1. Console error:
```
Uncaught TypeError: Cannot read property 'state' of undefined
  at t.handleTouchMove
```

Fix should be either to call `.bind` in the constructor (like you do for the other class methods) or use this syntax:
```js
handleTouchMove = (evt) => {
  // ...
};
```